src/connection.rs

```diff
@@ -545,9 +544,7 @@ impl rustls_connection {
     #[no_mangle]
     pub extern "C" fn rustls_connection_free(conn: *mut rustls_connection) {
         ffi_panic_boundary! {
-            let conn: &mut Connection = try_mut_from_ptr!(conn);
-            // Convert the pointer to a Box and drop it.
-            drop(unsafe { Box::from_raw(conn) });
+            free_box(conn);
```

Comment:
I think this was a spot where we missed using the `BoxCastPtr::to_box` helper previously, and had re-implemented it inline.